Modules/Quotas/test/Quotas.Domain.Tests/Tests/Identities/IdentityTests.cs

```diff
@@ -44,6 +45,52 @@ public void Can_assign_tier_quota_from_definition_to_identity()
         identity.TierQuotas.Second().Period.Should().Be(tierQuotaDefinition2.Period);
     }
 
+    [Fact]
+    public void Can_delete_tier_quota_by_definition_id()
+    {
+        // Arrange
+        var identity = new Identity("some-address", new TierId("some-tier-id"));
+        var tierQuotaDefinition = new TierQuotaDefinition(MetricKey.NumberOfSentMessages, 1, QuotaPeriod.Day);
+        identity.AssignTierQuotaFromDefinition(tierQuotaDefinition);
+
+        // Act
+        identity.DeleteTierQuotaFromDefinitionId(tierQuotaDefinition.Id);
+
+        // Assert
+        identity.TierQuotas.Should().HaveCount(0);
+    }
+
+    [Fact]
+    public void Can_delete_tier_quota_by_definition_id_with_multiple_quotas()
+    {
+        // Arrange
+        var identity = new Identity("some-address", new TierId("some-tier-id"));
+        var tierQuotaDefinition1 = new TierQuotaDefinition(MetricKey.NumberOfSentMessages, 1, QuotaPeriod.Day);
+        var tierQuotaDefinition2 = new TierQuotaDefinition(MetricKey.NumberOfSentMessages, 1, QuotaPeriod.Day);
+        identity.AssignTierQuotaFromDefinition(tierQuotaDefinition1);
+        identity.AssignTierQuotaFromDefinition(tierQuotaDefinition2);
+
+        // Act
+        identity.DeleteTierQuotaFromDefinitionId(tierQuotaDefinition1.Id);
+
+        // Assert
+        identity.TierQuotas.Should().HaveCount(1);
+        identity.TierQuotas.ElementAt(0).DefinitionId.Should().Be(tierQuotaDefinition2.Id);
+    }
+
+    [Fact]
+    public void Cannot_delete_nonexistent_tier_quota_by_definition_id()
+    {
+        // Arrange
+        var identity = new Identity("some-address", new TierId("some-tier-id"));
+
+        // Act
+        var acting = () => identity.DeleteTierQuotaFromDefinitionId("some-nonexistent-tier-quota-definition-id");
```

Comment:
```suggestion
        var acting = () => identity.DeleteTierQuotaFromDefinitionId(new TierQuotaDefinitionId("some-inexistent-id"));
```

Reply:
The `TierQuotaDefinitionId` constructor is private and the `TierQuotaDefinitionId.Create` method has a validation. Should I just use `TierQuotaDefinitionId.New()`?

Reply:
Oh, I forgot about that. In that case, you can go with `TierQuotaDefinitionId.Create("TIRsomeInexistentId")`